File: packages/lang-dotnet/src/Api/Models/TaskModel.cs
using System.ComponentModel.DataAnnotations;

namespace Api.Models
{
  public class TaskModel
  {
    public long Id { get; set; }

    [Required]
    public string Name { get; set; }

    public string Content { get; set; }

    [Required]
    public bool Finished { get; set; } = false;
  }
}

File: packages/db-marshall/src/task/task.model.ts
/* eslint-disable @typescript-eslint/no-inferrable-types */
import { Entity, uuid, f } from '@marcj/marshal'
import { TaskBase } from '@todo/lib-db'

@Entity('Task')
export class TaskModel implements TaskBase {
  @(f.primary().uuid())
  id: string = uuid()

  @f
  createdAt: Date = new Date()

  @f
  updatedAt: Date = new Date()

  constructor(
    @f public name: string,
    @f.optional() content?: string,
    @f public finished: boolean = false,
  ) {}
}
